frontend/src/components/administration/users/dialog-edit-user.tsx

```diff
@@ -1,148 +1,155 @@
-import FormControl from '@mui/material/FormControl';
+import { SelectFieldForm } from '@components/edit-profile/select-field-form/select-field-form';
+import { useAppForm } from '@hooks/hooks';
 import MenuItem from '@mui/material/MenuItem';
-import { SelectChangeEvent } from '@mui/material/Select';
+import Modal from '@mui/material/Modal';
 
-type DialogEditUserProps = {
-  user?: User;
-  handleClose: () => void;
-  handleSubmit: (user: User, newUserData: Partial<Omit<User, 'id'>>) => void;
-};
-
-const DialogEditUser: FC<DialogEditUserProps> = (props) => {
-  const { user, handleClose, handleSubmit } = props;
-  const [role, setRole] = useState<UserRole | undefined>(user?.role);
-  const [name, setName] = useState<string | undefined>(user?.name);
-  const [email, setEmail] = useState<string | undefined>(user?.email);
-  const [phone, setPhone] = useState<string | undefined>(user?.phone);
-  const [location, setLocation] = useState<string | undefined>(user?.location);
-  const [sex, setSex] = useState<UserSex | undefined>(user?.sex);
+import styles from './styles.module.scss';
 
-  useEffect(() => {
-    if (!user) return;
-    setName(user.name);
-    setEmail(user.email);
-    setPhone(user.phone);
-    setLocation(user.location);
-    setSex(user.sex);
-    setRole(user.role);
-  }, [user]);
+interface DialogEditUserProps {
+  user?: User;
+  onClose: () => void;
+  onSave: (user: User, newUserData: Partial<Omit<User, 'id'>>) => void;
+}
 
-  const handleChangeName = (event: ChangeEvent<HTMLTextAreaElement>): void => {
-    setName(event.target.value);
-  };
+type EditUserRequestData = Partial<User>;
 
-  const handleChangeEmail = (event: ChangeEvent<HTMLTextAreaElement>): void => {
-    setEmail(event.target.value);
-  };
+const DialogEditUser: FC<DialogEditUserProps> = (props) => {
+  const { user, onClose, onSave } = props;
 
-  const handleChangePhone = (event: ChangeEvent<HTMLTextAreaElement>): void => {
-    setPhone(event.target.value);
-  };
+  const { control, errors, handleSubmit, setValue, clearErrors } =
+    useAppForm<EditUserRequestData>({
+      defaultValues: {
+        sex: user?.sex || 'not_appliable',
+        location: user?.location,
+        name: user?.name,
+        phone: user?.phone,
+        email: user?.email,
+      },
+      validationSchema: editUserSchema,
+    });
 
-  const handleChangeLocation = (
-    event: ChangeEvent<HTMLTextAreaElement>,
-  ): void => {
-    setLocation(event.target.value);
+  const handleClose = (): void => {
+    if (clearErrors) clearErrors();
+    onClose();
   };
 
-  const handleChangeSex = (event: SelectChangeEvent<string>): void => {
-    setSex(event.target.value as unknown as UserSex);
-  };
+  useEffect(() => {
+    if (!user) return;
+    setValue('name', user.name);
+    setValue('email', user.email);
+    setValue('phone', user.phone);
+    setValue('location', user.location);
+    setValue('sex', user.sex);
+    setValue('role', user.role);
```

Comment:
This set of invocations can be simplified - https://stackoverflow.com/a/69536828/10963661

Reply:
Thanks. It's really a better way. Done.